src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -495,6 +495,17 @@ private unsafe bool StartWithCreateProcess(ProcessStartInfo startInfo)
                         startupInfo.dwFlags = Interop.Advapi32.StartupInfoOptions.STARTF_USESTDHANDLES;
                     }
 
+                    if (startInfo.WindowStyle != ProcessWindowStyle.Normal)
+                    {
+                        startupInfo.wShowWindow = startInfo.WindowStyle switch
+                        {
+                            ProcessWindowStyle.Minimized => (short)Interop.Shell32.SW_SHOWMINIMIZED,
+                            ProcessWindowStyle.Maximized => (short)Interop.Shell32.SW_SHOWMAXIMIZED,
+                            _ => (short)Interop.Shell32.SW_HIDE,
+                        };
```

Comment:
For maintainability, could you please move this into a `GetShowWindowFromWindowStyle(ProcessWindowStyle windowStyle)` helper and use it for this path and UseShellExecute path. https://github.com/dotnet/runtime/blob/66465ef8e7808b2c3e68a5e5a0a09cba2e96f9a8/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Win32.cs#L72-L78

Reply:
Sure